geomesa-tools/src/main/scala/org/locationtech/geomesa/tools/export/formats/LeafletMapExporter.scala

```diff
@@ -0,0 +1,185 @@
+      } else {
+        indexWriter.write(",")
+      }
+      json.writeFeature(feature, indexWriter)
+      storeFeature(feature, coordMap)
+      count += 1L
+    }
+    indexWriter.flush()
+    totalCount += count
+    Some(count)
+  }
+
+  override def close(): Unit  = {
+    // Finish writing GeoJson
+    indexWriter.write("]};\n\n")
+    indexWriter.write(getFeatureInfo(sft))
+    // Write Heatmap Data
+    val values = normalizeValues(coordMap)
+    first = true
+    indexWriter.write("""var heat = L.heatLayer([""" + "\n")
+    values.foreach{ c =>
+      if (first) {
+        first = false
+      } else {
+        indexWriter.write(",\n")
+      }
+      indexWriter.write(c._1.render(c._2))
+    }
+    indexWriter.write("\n    ], {radius: 25});\n\n")
+    indexWriter.write(indexTail)
+    indexWriter.close()
+
+    if (totalCount < 1) user.warn("No features were exported. " +
+      "This will cause the map to fail to render correctly.")
+    // Use println to ensure we write the destination to stdout so the bash wrapper can pick it up.
+    System.out.println("\nSuccessfully wrote Leaflet html to: " + indexFile.toString + "\n")
+  }
+}
+
+object LeafletMapExporter {
+  def checkDestination(file: File): File = {
+    try {
+      file.mkdir()
+      if (! file.isDirectory) {
+        throw new RuntimeException(s"Output destination ${file.toString} must not exist or must be a directory.")
+      }
+      file
+    } catch {
+      case e: SecurityException => throw new RuntimeException("Unable to create output destination, check permissions.", e)
+    }
+  }
+
+  def getFeatureInfo(sft: SimpleFeatureType): String = {
+    val str: StringBuilder = new StringBuilder()
+    str.append("    function onEachFeature(feature, layer) {\n")
+    Option(sft) match {
+      case None    => str.append("\n    };\n\n").toString
+      case Some(_) =>
+        str.append("""        layer.bindPopup("ID: " + feature.id + "<br>" + """)
+        str.append(""""GEOM: " + feature.geometry.type + "[" + feature.geometry.coordinates + "]<br>" + """)
+        str.append(sft.getAttributeDescriptors.filter(_.getLocalName != "geom") .map(attr =>
+          s""""${attr.getLocalName}: " + feature.properties.${attr.getLocalName}"""
+        ).mkString(""" + "<br>" + """))
+
+    }
+    str.append(");\n    }\n\n")
+    str.toString()
+  }
+
+  def storeFeature(feature: SimpleFeature, coordMap: mutable.Map[SimpleCoordinate[Double], Int]): Unit = {
+    val coords: Array[Coordinate] = feature.getDefaultGeometry match {
+      case geom: Geometry => geom.getCoordinates
+      case _ => Array[Coordinate]()
+    }
+    coords.map(c => SimpleCoordinate(c.x, c.y)).foreach{ sc =>
+      coordMap.put(sc, 1) match {
+        case Some(count) => coordMap.put(sc, count + 1)
+        case None =>
+      }
+    }
+  }
+
+  def normalizeValues(coordMap: mutable.Map[SimpleCoordinate[Double], Int]): Map[SimpleCoordinate[Double], Float] = {
+    val max: Float = coordMap.maxBy(_._2)._2
```

Comment:
probably need to handle the case here where no features are returned - think this would throw an exception

Reply:
Good catch.